subprojects/file-watching/src/main/java/org/gradle/internal/watch/vfs/impl/WatchingVirtualFileSystem.java

```diff
@@ -165,11 +168,17 @@ public void registerWatchableHierarchy(File watchableHierarchy) {
             }
             return withWatcherChangeErrorHandling(
                 currentRoot,
-                () -> watchRegistry.registerWatchableHierarchy(watchableHierarchy, currentRoot)
+                () -> registerWatchableHierarchyInternal(currentRoot, watchableHierarchy)
             );
         });
     }
 
+    private void registerWatchableHierarchyInternal(SnapshotHierarchy currentRoot, File watchableHierarchy) {
+        // TODO Do not check if file system watching was enabled explicitly
+        watchableFileSystemRegistry.ensureWatchingSupported(watchableHierarchy);
```

Comment:
We shouldn't fail here if the hierarchy is not supported by default, right? We should only watch hierarchies which we can watch.